### PR TITLE
IDs for adjuntos must be formatted as integer

### PR DESCRIPTION
**Pull request by seppo0010**
Skip missing keys as they crash when asserting they are floats.

### CODE REVIEW

File: builder/crawler/lib.go [hunk 1]
```diff
@@ -206,24 +206,30 @@ func GetAdjuntosCedula(url string, ficha *shared.Ficha, actuacion *shared.Actuac
...
+		if val, found := adjunto["adjuntoId"]; !found || val == nil {
+			log.WithFields(log.Fields{
+				"adjunto": adjunto,
+			}).Warn("skipping adjunto")
+			continue
+		}
 		url := fmt.Sprintf("https://eje.juscaba.gob.ar/iol-api/api/public/expedientes/cedulas/adjuntoPdf?filter=%%7B%%22aacId%%22:%v,%%22expId%%22:%v,%%22ministerios%%22:false%%7D",
-			adjunto["adjuntoId"],
+			int(adjunto["adjuntoId"].(float64)),
 			ficha.ExpId,
```
Comment (xaiki) [inline builder/crawler/lib.go:219]: Looks strange, why do float64 at all ?

File: builder/crawler/lib.go [hunk 2]
```diff
@@ -206,24 +206,30 @@ func GetAdjuntosCedula(url string, ficha *shared.Ficha, actuacion *shared.Actuac
...
 		)
-		documentos[i] = &shared.Documento{
+		documentos = append(documentos, &shared.Documento{
 			URL:                url,
```
Comment (xaiki) [inline builder/crawler/lib.go:222]: Should go to another commit ?

File: builder/crawler/lib.go [hunk 3]
```diff
@@ -206,24 +206,30 @@ func GetAdjuntosCedula(url string, ficha *shared.Ficha, actuacion *shared.Actuac
...
 			Nombre:             adjunto["adjuntoNombre"].(string),
-		}
+		})
 	}
 	return documentos, nil
 }
 func GetAdjuntosNoCedula(url string, ficha *shared.Ficha, actuacion *shared.Actuacion) ([]*shared.Documento, error) {
-	u := fmt.Sprintf("https://eje.juscaba.gob.ar/iol-api/api/public/expedientes/actuaciones/adjuntos?actId=%v&expId=%v&accesoMinisterios=false",
+	u := fmt.Sprintf("https://eje.juscaba.gob.ar/iol-api/api/public/expedientes/actuaciones/adjuntos?actId=%d&expId=%v&accesoMinisterios=false",
 		actuacion.ActId,
```
Comment (xaiki) [inline builder/crawler/lib.go:233]: Why isn't this line enough ?

File: builder/crawler/lib.go [hunk 4]
```diff
@@ -206,24 +206,30 @@ func GetAdjuntosCedula(url string, ficha *shared.Ficha, actuacion *shared.Actuac
...
 	}
-	documentos := make([]*shared.Documento, len(adjuntos))
-	for i, adjunto := range adjuntos {
+	documentos := make([]*shared.Documento, 0, len(adjuntos))
+	for _, adjunto := range adjuntos {
+		if val, found := adjunto["adjuntoId"]; !found || val == nil {
+			log.WithFields(log.Fields{
+				"adjunto": adjunto,
+			}).Warn("skipping adjunto")
+			continue
```
Comment (xaiki) [inline builder/crawler/lib.go:215]: Log should go to another commit

File: builder/crawler/lib.go [hunk 5]
```diff
@@ -247,19 +253,25 @@ func GetAdjuntosNoCedula(url string, ficha *shared.Ficha, actuacion *shared.Actu
...
-	documentos := make([]*shared.Documento, len(adjuntos["adjuntos"]))
-	for i, adjunto := range adjuntos["adjuntos"] {
+	documentos := make([]*shared.Documento, 0, len(adjuntos["adjuntos"]))
+	for _, adjunto := range adjuntos["adjuntos"] {
+		if val, found := adjunto["adjId"]; !found || val == nil {
+			log.WithFields(log.Fields{
+				"adjunto": adjunto,
+			}).Warn("skipping adjunto")
+			continue
+		}
```
Comment (xaiki) [inline builder/crawler/lib.go:263]: Ditto

File: builder/crawler/lib.go [hunk 6]
```diff
@@ -247,19 +253,25 @@ func GetAdjuntosNoCedula(url string, ficha *shared.Ficha, actuacion *shared.Actu
...
 		url := fmt.Sprintf("https://eje.juscaba.gob.ar/iol-api/api/public/expedientes/actuaciones/adjuntoPdf?filter=%%7B%%22aacId%%22:%v,%%22expId%%22:%v,%%22ministerios%%22:false%%7D",
-			adjunto["adjId"],
+			int(adjunto["adjId"].(float64)),
 			ficha.ExpId,
 		)
-		documentos[i] = &shared.Documento{
+		documentos = append(documentos, &shared.Documento{
 			URL:                url,
```
Comment (xaiki) [inline builder/crawler/lib.go:269]: Ditto

**Review by xaiki — CHANGES_REQUESTED** [review]

File: builder/crawler/lib.go (hunk 1 above)
Comment (seppo0010) [reply builder/crawler/lib.go:219]: json parse uses float64

File: builder/crawler/lib.go (hunk 2 above)
Comment (seppo0010) [reply builder/crawler/lib.go:222]: it is related to the continue that's a consequence of the float64 change

Comment (seppo0010) [reply builder/crawler/lib.go:219]: (casting nil to float64 is a fatal error)

File: builder/crawler/lib.go (hunk 1 above)
Comment (seppo0010) [reply builder/crawler/lib.go:216]: `var.(type)` is a "type assertion" while `type(var)` is casting
the map can contain any type of variable, I'm saying `adjuntoId` must be `float64` and then casting to `int`

File: builder/crawler/lib.go (hunk 3 above)
Comment (xaiki) [reply builder/crawler/lib.go:230]: Casting ? Anyway i don't go, if you think it's ok like that, just merge
